swap/swap.go

```diff
@@ -334,12 +338,9 @@ func (s *Swap) loadBalance(peer enode.ID) (err error) {
 // sendCheque sends a cheque to peer
 // To be called with mutex already held
 // Caller must be careful that the same resources aren't concurrently read and written by multiple routines
-func (s *Swap) sendCheque(peer enode.ID) error {
-	swapPeer, ok := s.getPeer(peer)
-	if !ok {
-		return fmt.Errorf("error while getting peer: %s", peer)
-	}
-	cheque, err := s.createCheque(peer)
+func (s *Swap) sendCheque(swapPeer *Peer) error {
+	peer := swapPeer.Peer.ID()
```

Comment:
You should be able to just call `swapPeer.ID()` (`Peer` is embedded)

Reply:
fixed, thanks